rocketpy/EnvironmentAnalysis.py

```diff
@@ -588,22 +594,99 @@ def calculate_maximum_wind_gust(self):
         self.max_wind_gust = np.max(self.wind_gust_list)
         return self.max_wind_gust
 
-    # TODO: Implement
+    # TODO: Create tests
     def calculate_wind_gust_distribution(self):
         """Get all values of wind gust speed (for every date and hour available)
-        and plot a single distribution. Expectedr result is a Weibull distribution.
+        and plot a single distribution. Expected result is a Weibull distribution.
         """
-        ...
+        self.wind_gust_list = [
+            dayDict[hour]["surfaceWindGust"]
+            for dayDict in self.surfaceDataDict.values()
+            for hour in dayDict.keys()
+        ]
+        plt.figure()
+        # Plot histogram
+        plt.hist(
+            self.wind_gust_list,
+            bins=int(len(self.wind_gust_list) ** 0.5),
+            density=True,
+            histtype="stepfilled",
+            alpha=0.2,
+            label="Wind Gust Speed Distribution",
+        )
+
+        # Plot weibull distribution
+        c, loc, scale = scipy.stats.weibull_min.fit(self.wind_gust_list)
+        x = np.linspace(0, np.max(self.wind_gust_list), 100)
+        plt.plot(
+            x,
+            scipy.stats.weibull_min.pdf(x, c, loc, scale),
+            "r-",
+            linewidth=2,
+            label="Weibull Distribution",
+        )
+
+        # Label plot
+        plt.ylabel("Probability")
+        plt.xlabel("Wind gust speed (m/s)")
+        plt.title("Wind Gust Speed Distribution")
+        plt.legend()
+        plt.show()
+
+        return self.wind_gust_list
 
     # TODO: Implement
     def calculate_average_temperature_along_day(self):
         """temperature progression throughout the day at some fine interval (ex: 2 hours) with 1, 2, 3, sigma contours"""
         ...
 
-    # TODO: Implement
-    def calculate_average_wind_profile(self):
-        """average, 1, 2, 3 sigma wind profile from 0 35,000 ft AGL"""
-        ...
+    # TODO: Create tessts
+    def calculate_average_wind_speed_profile(self):
```

Comment:
I'm receiving this error when running this function:

`TypeError: _percentile_dispatcher() got an unexpected keyword argument 'method'`

Reply:
I will try to reproduce this error here. Thanks for catching it.

Reply:
As reported here (https://github.com/numpy/numpy/issues/21184) this is a compatibility problem. Numpy versions before 1.22.0 do not support the `method` attribute in `np.percentile`. This is not a problem in most cases, so I just removed it and left the default linear behavior.
Thanks again for catching this @PatrickSampaioUSP.